src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -247,6 +243,16 @@ def _count_processed_transactions(self, org_id: int, bucket_payload: MetricsBuck
             # Unexpected value type for this metric ID, skip.
             return 0
 
+    def _get_metric_name(self, bucket_payload: MetricsBucket) -> Optional[str]:
+        metric_id = str(bucket_payload["metric_id"])
+
+        for mapping in bucket_payload.get("mapping_meta", {}).values():
+            for id, name in mapping.items():
+                if id == metric_id:
+                    return name
+
+        return None
```

Comment:
Did not test this code, but the outer key in `mapping_meta` is the org_id IIRC.

```suggestion
        org_id = bucket_payload["org_id"]
        metric_id = str(bucket_payload["metric_id"])
        
        return bucket_payload.get("mapping_meta", {}).get(org_id, {}).get(metric_id)
```

TBH I would even use `[]` instead of `.get()`, and log an error when a `KeyError` is raised. If I understand correctly, an incomplete `mapping_meta` would be a bug that we want to know about.

Reply:
I'm not sure if it's the org_id; and if it is, it may be because of the current implementation. Running my local dev env, a bucket payload looks like the following:

<details>
<summary>Metric bucket payload, in my local dev env</summary>

```json
{
  "org_id": 1,
  "project_id": 1,
  "type": "d",
  "value": [29.751],
  "timestamp": 1669029110,
  "tags": {
    "9223372036854776010": 65536,
    "9223372036854776035": 9223372036854776037,
    "65547": 65541,
    "9223372036854776016": 65681,
    "9223372036854776026": 9223372036854776031,
    "9223372036854776020": 65610,
    "9223372036854776022": 65638,
    "9223372036854776021": 9223372036854776027
  },
  "metric_id": 9223372036854775909,
  "retention_days": 90,
  "mapping_meta": {
    "h": {
      "9223372036854776010": "environment",
      "9223372036854776026": "satisfaction",
      "9223372036854775909": "d:transactions/duration@millisecond",
      "9223372036854776020": "transaction",
      "9223372036854776037": "inlier",
      "9223372036854776016": "release",
      "9223372036854776027": "ok",
      "9223372036854776021": "transaction.status",
      "9223372036854776031": "satisfied",
      "9223372036854776022": "transaction.op",
      "9223372036854776035": "histogram_outlier"
    },
    "d": {
      "65536": "development",
      "65541": "python",
      "65681": "backend@22.11.0.dev0+3184b29af3bc4dcc33f98ca420f9ec5042eb7206",
      "65610": "sentry.snuba.tasks.subscription_checker",
      "65547": "platform",
      "65638": "queue.task.celery"
    }
  },
  "use_case_id": "performance"
}
```

</details>

I think it's safer to not make any assumptions here (especially considering the reason for this change is to not make assumptions on the indexer's internal implementation). I agree on raising an error though, following up on this.